src/vcpkg-test/cmake.cpp

```diff
@@ -0,0 +1,74 @@
+#include <vcpkg-test/util.h>
+
+#include <vcpkg/cmake.h>
+
+using namespace vcpkg;
+
+TEST_CASE ("replace CMake variable", "[cmake]")
+{
+    static constexpr StringLiteral str{"lorem ip${VERSION}"};
+    {
+        auto res = replace_cmake_var(str, "VERSION", "sum");
+        REQUIRE(res == "lorem ipsum");
+    }
+    {
+        auto res = replace_cmake_var(str, "VERSiON", "sum");
+        REQUIRE(res == "lorem ip${VERSION}");
+    }
+}
+
+TEST_CASE ("find cmake invocation", "[cmake]")
+{
+    {
+        auto res = find_cmake_invocation("lorem_ipsum()", "lorem_ipsum");
+        REQUIRE(res.empty());
+    }
+    {
+        auto res = find_cmake_invocation("lorem_ipsum()", "lorem_ipsu");
+        REQUIRE(res.empty());
+    }
+    {
+        auto res = find_cmake_invocation("lorem_ipsum(", "lorem_ipsum");
+        REQUIRE(res.empty());
+    }
+    {
+        auto res = find_cmake_invocation("lorem_ipum()", "lorem_ipsum");
+        REQUIRE(res.empty());
+    }
+    {
+        auto res = find_cmake_invocation("lorem_ipsum( )", "lorem_ipsum");
```

Comment:
This should test additional edge cases:
1. `lorem_ipsum_`
2. `lorem_ipsum (`
3. `lorem_ipsum` (EOF)